baseclass/test/java/com/google/apigee/calloutbaseclass/CalloutBaseTest.java

```diff
@@ -0,0 +1,225 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * <p>Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file
+ * except in compliance with the License. You may obtain a copy of the License at
+ *
+ * <p>https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * <p>Unless required by applicable law or agreed to in writing, software distributed under the
+ * License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.apigee.calloutbaseclass;
+
+import com.apigee.flow.message.MessageContext;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+public final class CalloutBaseTest {
+
+  private static final String ERROR_FLOW_VARIABLE = "callout_error";
+  private static final String EXCEPTION_FLOW_VARIABLE = "callout_exception";
+  private static final String EXCEPTION_STACKTRACE_FLOW_VARIABLE = "callout_exception_stacktrace";
+  private static final String LOG_FLOW_VARIABLE = "callout_log";
+  private static final String NONEXISTENT_VARIABLE_REFERENCE = "{fake.variable}";
+  private static final String OPTIONAL_VARIABLE_KEY = "optionalVar";
+  private static final String OPTIONAL_VARIABLE_VALUE = "optionalValue";
+  private static final String OPTIONAL_VARIABLE_EMPTY_VALUE_KEY = "optionalVarEmpty";
+  private static final String REQUIRED_VARIABLE_KEY = "requiredVar";
+  private static final String REQUIRED_VARIABLE_VALUE = "requiredValue";
+  private static final String REQUIRED_VARIABLE_EMPTY_VALUE_KEY = "requiredVarEmpty";
+  private static final String TEST_FLOW_VALUE = "test123";
+  private static final String TEST_FLOW_VARIABLE = "request.queryparam.testFlowVar";
+  private static final String TEST_FLOW_VARIABLE_REFERENCE = "{request.queryparam.testFlowVar}";
+  private static final String TEST_LOG_STATEMENT = "asdf123";
+  private static final String TEST_LOG_STATEMENT2 = "defgh5678";
+  @Spy FakeMessageContext messageContext;
```

Comment:
I assume this cannot be private because of @Spy?

Reply:
actually it can be. changed to private.